Azure IOT Central updates for DotNet SDK release for RP version 2021-06-01

gansvv opened the pull request:
# Changes Included

This release removes legacy S1 SKU, 
        adds support for regions while removing geographies (for location), 
        releases new RP version 2021-06-01,
        and adds support system-assigned Managed Identities in IoT Central.

# All SDK Contribution checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.
- [X] **Please open PR in `Draft` mode if it is:**
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.
- [X] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [X] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md).**
- [X] **The pull request does not introduce [breaking changes](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md).**

### [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#general-guidelines)
- [X] Title of the pull request is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#testing-guidelines)
- [X] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#sdk-generation-guidelines)
- [X] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code. (Track 2 only)
- [X] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK. Please double check nuget.org current release version.

## Additional management plane SDK specific contribution checklist: 
Note: Only applies to `Microsoft.Azure.Management.[RP]` or `Azure.ResourceManager.[RP]`
 
- [X] Include updated [management metadata](https://github.com/Azure/azure-sdk-for-net/tree/main/eng/mgmt/mgmtmetadata).
- [X] Update AzureRP.props to add/remove version info to maintain up to date API versions.

### Management plane SDK Troubleshooting
- If this is very first SDK for a services and you are adding new service folders directly under /SDK, please add `new service` label and/or contact assigned reviewer.
- If the check fails at the `Verify Code Generation` step, please ensure:
	- Do not modify any code in generated folders.
	- Do not selectively include/remove generated files in the PR.
	- Do use `generate.ps1/cmd` to generate this PR instead of calling `autorest` directly.
	Please pay attention to the @microsoft.csharp version output after running `generate.ps1`. If it is lower than current released version (2.3.82), please run it again as it should pull down the latest version.
	
	**Note: We have recently updated the PSH module called by `generate.ps1` to emit additional data. This would help reduce/eliminate the Code Verification check error. Please run following command**:

	    `dotnet msbuild eng/mgmt.proj /t:Util /p:UtilityName=InstallPsModules`

### Old outstanding PR cleanup
 Please note:
	If PRs (including draft) has been out for more than 60 days and there are no responses from our query or followups, they will be closed to maintain a concise list for our reviewers.

gansvv committed:
Updated for RP 2021-06-01 release

This release removes legacy S1 SKU, has S360 fixes, releases new RP version 2021-06-01,
        and adds support for regional resources and system-assigned Managed Identities in IoT Central.

gansvv committed:
Updated release notes, added tests for Managed Identities and different SKUs.

Updated package release notes,
added tests for Managed Identities,
added tests for validating exceptions from invalid SKUs.

poanchen reviewed (approved):
LGTM

gansvv committed:
Minor: Updated default Sku in test

Updated default sku to be ST2.

archerzz reviewed (changes requested):
@gansvv  There are a few constants deleted from the API, which is a breaking change. We need to bump the version to `4.0.0`. Others looks fine. Thanks.

gansvv committed:
Updated version to 4.0.0

Updated version to 4.0.0 based on PR feedback.

gansvv commented:
> @gansvv There are a few constants deleted from the API, which is a breaking change. We need to bump the version to `4.0.0`. Others looks fine. Thanks.

Thanks, I've updated version to 4.0.0. cc @archerzz

archerzz commented:
@gansvv Pipeline is broken due to test failure. It seems that the test recordings are obsolete. Could you re-record the tests? Thanks.

gansvv committed:
Adding Session Records from execution of the IoTCentral DotNet SDK Tests

Adding Session Records from execution of the IoTCentral DotNet SDK Tests

gansvv commented:
> @gansvv Pipeline is broken due to test failure. It seems that the test recordings are obsolete. Could you re-record the tests? Thanks.

Done. Added to PR.

gansvv committed:
Remove unused debug project ref to TestFramework

Remove debug project ref to TestFramework

gansvv committed:
Removed randomization during session recording to prevent pipeline build failures for tests

Removed randomization during session recording to prevent pipeline build failures for tests

gansvv committed:
Removed randomization from UpdateLIfeCycle tests and re-ran session records

Removed randomization from UpdateLIfeCycle tests and re-ran session records

gansvv committed:
Revert change to mgmtcommon file

Revert change to mgmtcommon file.

https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/mgmtcommon/TestFramework/TestRecordAndFramework.sln

gansvv committed:
Minor update to sln

Minor update to match raw doc from Azure repo.

gansvv committed:
Add comment to trigger new build

Add comment to trigger new build

gansvv committed:
Increasing test pipeline timeout to 100mins since most builds are failing after 60mins.

Increasing test pipeline timeout to 100mins since most builds are failing after 60mins. Matched build timeout in pipeline config.

archerzz reviewed (approved):
The codes look good. Please roll back the change in `ci.mgmt.yml` and `packge-lock.json` which will have impact on others as well.

gansvv committed:
Undo changes to test pipeline timeout and package-lock.json

Undo changes to test pipeline timeout and package-lock.json after syncing with Mingzhe Huang.

gansvv commented:
> The codes look good. Please roll back the change in `ci.mgmt.yml` and `packge-lock.json` which will have impact on others as well.

Done.

gansvv committed:
Minor: update whitespace to match package-lock.json

Minor: updated whitespace to match package-lock.json

archerzz commented:
/check-enforcer override

archerzz commented:
The test actually passes. The failure is due to timeout in macOS node, which should be fixed in pipeline.
